Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/CosmosServerlessSparkBatchJob.java

```diff
@@ -152,7 +146,13 @@ public String getJobUuid() {
     @Override
     public Observable<SparkBatchJobResponsePayload> getStatus() {
         return getAccount().getSparkBatchJobRequest(getJobUuid())
-                .map(sparkBatchJob -> sparkBatchJob.properties().responsePayload());
+                .flatMap(sparkBatchJob -> {
+                    if (sparkBatchJob.properties() == null) {
+                        return Observable.error(new IOException("Can't get job status after retry " + getRetriesMax() + " times."));
```

Comment:
```suggestion
                        return Observable.error(new IOException("Can't get job status from empty property body."));
```

Let the caller to handle the retry error.